tests/integration/components/denali-progress-test.js

```diff
@@ -0,0 +1,166 @@
+  });
+
+  test('it supports color and shade', async function (assert) {
+    await render(hbs`
+      <DenaliProgress @value={{this.value}} @color={{this.color}} @shade={{this.shade}} />
+    `);
+
+    this.set('value', 30);
+    assert
+      .dom('.d-progress div')
+      .hasAttribute('aria-valuenow', '30', 'DenaliProgress with default color should be displayed');
+
+    this.set('color', 'red');
+    assert
+      .dom('div.d-progress--bar')
+      .hasClass(
+        'has-bg-red-500',
+        'DenaliProgress displayed with red color when `@color` arg is set to red and default shade value 500'
+      );
+
+    this.set('shade', '100');
+    assert
+      .dom('div.d-progress--bar')
+      .hasClass(
+        'has-bg-red-100',
+        'DenaliProgress displayed with red color and shade 100 when `@color` arg is set to red and `@shade` is set to 100'
+      );
+
+    this.set('color', 'green');
+    this.set('shade', '300');
+    assert
+      .dom('div.d-progress--bar')
+      .hasClass(
+        'has-bg-green-300',
+        'DenaliProgress displayed with green color and shade 300 when `@color` arg is set to green and `@shade` is set to 300'
+      );
+  });
+
+  test('it supports isLoading', async function (assert) {
+    await render(hbs`
+      <DenaliProgress @isLoading={{this.isLoading}} @size={{this.size}}/>
+    `);
+
+    assert.dom('div.d-progress').exists('DenaliProgress can render with default `@isLoading` arg set to false ');
+    this.set('isLoading', true);
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__loading', 'DenaliProgress has a loading class when `@isLoading` arg is set to true');
+
+    this.set('size', 'small');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__sm', 'DenaliProgress has a smaill size when `@size` arg is set to small');
+  });
+
+  test('it supports sizes', async function (assert) {
+    await render(hbs`
+      <DenaliProgress @size={{this.size}}/>
+    `);
+
+    assert.dom('div.d-progress').exists('DenaliProgress can render with default size');
+
+    this.set('size', 'small');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__sm', 'DenaliProgress has a smaill size when `@size` arg is set to small');
+
+    this.set('size', 'medium');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__md', 'DenaliProgress has a medium size when `@size` arg is set to medium');
+
+    this.set('size', 'large');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__lg', 'DenaliProgress has a large size when `@size` arg is set to large');
+  });
+
+  test('it supports multi level progress with colors', async function (assert) {
+    await render(hbs`
+    <DenaliProgress
+      @size={{size}}
+      @isLoading={{isLoading}}
+      as |Progress|
+      >
+      <Progress.Bar @color="red" @value={{10}} @shade="100" />
```

Comment:
Can we change it to be this to test that the bars take the top level values as defaults
```suggestion
      @isLoading={{isLoading}}
      @color="red"
      @shade="100"
      as |Progress|
      >
      <Progress.Bar @value={{10}} />
```